apps/user-group-ms/core/src/main/java/it/pagopa/selfcare/user_group/core/UserGroupServiceImpl.java

```diff
@@ -160,6 +174,8 @@ public UserGroupOperations updateGroup(String id, UserGroupOperations group) {
         if (UserGroupStatus.SUSPENDED.equals(foundGroup.getStatus())) {
             throw new ResourceUpdateException(TRYING_TO_MODIFY_SUSPENDED_GROUP);
         }
+        checkNameUniqueness(group.getName(), foundGroup.getProductId(), foundGroup.getInstitutionId());
```

Comment:
In the `updateGroup` method, when checking for name uniqueness, the current group being updated is not excluded from the search. This can falsely trigger a uniqueness violation if the group name remains unchanged. Suppose you have a group named "Engineering". When updating this group without changing its name, the checkNameUniqueness method will find the group itself and throw a ResourceAlreadyExistsException, preventing a valid update.

Reply:
You are right, I will fix this case implementing this [suggestion](https://github.com/pagopa/selfcare-user/pull/216#discussion_r1830835242)